internal/facade/client.go

```diff
@@ -88,6 +99,135 @@ type SpaceClient interface {
 	DeleteBinding(ctx context.Context, guid string) error
 
 	FindServicePlan(ctx context.Context, serviceOfferingName string, servicePlanName string, spaceGuid string) (string, error)
+
+	//TODO: Add methods for managing service keys
+	// AddInstanceInCache(key string, instance *Instance)
+	// GetInstanceFromCache(key string) (*Instance, bool)
+	// AddBindingInCache(key string, binding *Binding)
+	// GetBindingFromCache(key string) (*Binding, bool)
+}
+
+type SpaceClientBuilder func(string, string, string, string, config.Config) (SpaceClient, error)
+
+// Cache is a simple in-memory cache to store spaces, instances, and bindings
+type Cache struct {
+	spaces                 map[string]*Space
+	instances              map[string]*Instance
+	bindings               map[string]*Binding
+	mutex                  sync.RWMutex
+	lastCacheTime          time.Time
+	cacheTimeOut           time.Duration
```

Comment:
I am not sure if we might want to have different timeouts depending on the resource type